### PR TITLE
BracesFixer - Add allow_single_line_closure configuration

### PR DESCRIPTION
**Pull request by keradus**
Closing #2413

### CODE REVIEW

File: src/Fixer/Basic/BracesFixer.php [hunk 1]
```diff
@@ -228,6 +276,22 @@ function ($item) {
...
+                && $token->isGivenKind(T_FUNCTION)
+                && $tokensAnalyzer->isLambda($index)
+            ) {
+                $braceEndIndex = $tokens->findBlockEnd(
+                    Tokens::BLOCK_TYPE_CURLY_BRACE,
+                    $tokens->getNextTokenOfKind($index, array('{'))
+                );
+
+                if (!$this->isMultilined($tokens, $index, $braceEndIndex)) {
+                    $index = $braceEndIndex;
```
Comment (SpacePossum) [inline src/Fixer/Basic/BracesFixer.php]: missing blank line after this statement

**Comment by SpacePossum** [comment]
Looking good! :+1:

**Comment by fabpot** [comment]
👍

File: src/Fixer/Basic/BracesFixer.php [hunk 2]
```diff
@@ -26,9 +28,42 @@
...
-final class BracesFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+final class BracesFixer extends AbstractFixer implements ConfigurableFixerInterface, WhitespacesAwareFixerInterface
 {
     /**
+     * @var array
+     */
+    private $configuration;
+
+    private static $defaultConfiguration = array(
+        'allow_oneline_lambda' => false,
```
Comment (keradus) [inline src/Fixer/Basic/BracesFixer.php]: `allow_oneline_lambda` or `allow_one_line_lambda` ?

Comment (SpacePossum) [inline src/Fixer/Basic/BracesFixer.php]: `allow_single_line_closure` ^_^

**Comment by fabpot** [comment]
Any chance to get this one merged?

**Comment by SpacePossum** [comment]
Pretty big chance, however there is a limited team working (/merging) on this repo so please be patient :)

File: src/Fixer/Basic/BracesFixer.php (hunk 2 above)
Comment (keradus) [inline src/Fixer/Basic/BracesFixer.php]: thanks, renamed

File: src/Fixer/Basic/BracesFixer.php [hunk 3]
```diff
@@ -26,9 +28,42 @@
...
+     * @var array
+     */
+    private $configuration;
+
+    private static $defaultConfiguration = array(
+        'allow_single_line_closure' => false,
+    );
+
+    /**
+     * @param string[]|null $configuration
```
Comment (gharlan) [inline src/Fixer/Basic/BracesFixer.php]: `string[]` is wrong, values are bools

File: tests/Fixer/Basic/BracesFixerTest.php [hunk 4]
```diff
@@ -1714,6 +1714,55 @@ public function providePreserveLineAfterControlBrace()
      * @param string      $expected
      * @param null|string $input
      *
+     * @dataProvider provideFixWithAllowOnelineLambdaCases
+     */
+    public function testFixWithAllowOnelineLambda($expected, $input = null)
```
Comment (gharlan) [inline tests/Fixer/Basic/BracesFixerTest.php]: `testFixWithAllowSingleLineClosure`

File: src/Fixer/Basic/BracesFixer.php (hunk 3 above)
Comment (keradus) [inline src/Fixer/Basic/BracesFixer.php]: note: `array<string, bool>|null`

File: tests/Fixer/Basic/BracesFixerTest.php (hunk 4 above)
Comment (keradus) [inline tests/Fixer/Basic/BracesFixerTest.php]: done

File: src/Fixer/Basic/BracesFixer.php (hunk 3 above)
Comment (keradus) [inline src/Fixer/Basic/BracesFixer.php]: done

**Comment by keradus** [comment]
merged

**Comment by SpacePossum** [comment]
thanks @keradus :)

happy days @fabpot :)

**Pull request closed without merging**